### ISSUE TITLE
enable fullscreen recording

### ISSUE THREAD

**Issue by SetsunaKen** [phw/peek#61]
It would be nice to record the whole screen.

**Comment by orschiro** [comment]
Is there a technical limitation to recording full-screen? 

Can't we put the application into full-screen mode with F11 like many other applications?

**Comment by phw** [comment]
Just a heads up that you already can do this (at least on GNOME, you have to check if other environments also support toggling the fullscreen mode). To do so in GNOME:

1, Open GNOME Control Center, got to Keyboard settings and search for "Toggle fullscreen mode". If not defined, set a keyboard shortcut for it (I used `Super`+ `F12`)
2. In Peek make sure you have a keyboard shortcut set to start recording (default is `Ctrl` + `Alt` + `r`).
3. Using the configured keyboard shortcuts first Toggle Peek into fullscreen mode the start recording
4. To stop recording use Peek's keyboard shortcut again, get out of fullscreen again.

Of course this is something Peek could do on its own. My idea currently is that we add a "Fullscreen" checkbox to the new format selection popover. If checked Peek will switch into full screen mode before starting to record, and it will automatically unfullscreen when recording stops (so you get the UI back). 

Of course this requires that Peek can set a global keyboard shortcut to stop recording. That means it is built with libkeybinder and the shortcut can be registered (doesn't conflict with any other global shortcut). I would just disable the fullscreen option if no keyboard shortcut is configured.

**Comment by slabua** [comment]
This way, though, does not allow to record the gnome-shell interface.

**Comment by phw** [comment]
@slabua That's true, not sure what to do about this :(

**Comment by slabua** [comment]
@phw Yes strange, ffmpeg by itself is able to record the gnome-shell. Just my feeling, it looks like in peek it's somehow recording on a layer right below the shell (if such layered architecture even exists), because it does record the wallpaper beneath indeed.

**Comment by JianwenLi-GS** [comment]
It would be nice if peek can record top bar of the gnome shell.

**Comment by ElijahLynn** [comment]
Was super excited to use Peek, as it is so simple. But yeah, just went to record my first GIF of the gnome power down settings in the top right and realized I cannot. Would really welcome this capability.

**Comment by najcardboyz** [comment]
I think so, peek is a great tool for making GIF, it's great if you can record gnome-shell panels

**Comment by lbarron-md** [comment]
Dog food all the things!
![Peek 2019-09-24 12-05](https://user-images.githubusercontent.com/53449884/65529140-99b0ca80-dec3-11e9-9826-3a3ab0c80fd8.gif)

**Comment by Raymo111** [comment]
I love peek, it's so simple. However, I can't seem to full screen it. I'm on KDE. maximize works, but fullscreen does not. If I manually resize the window in peek to 3840x2160, it crashes. If I set it to 3830x2150, it doesn't crash, but defaults to the maximized dimensions of 3838x2025. Very frustrating. So far I've just mitigated this issue by shrinking whatever window I want to record, but I'd love to be able to record my topbar some day :)

**Comment by andersonid** [comment]
Still cant use this app 'cause doesnt have full screen option. 
How can I help you developers to achieve this?

**Comment by PabloDinella** [comment]
> Just a heads up that you already can do this (at least on GNOME, you have to check if other environments also support toggling the fullscreen mode). To do so in GNOME:
> 
> 1. Open GNOME Control Center, got to Keyboard settings and search for "Toggle fullscreen mode". If not defined, set a keyboard shortcut for it (I used `Super`+ `F12`)
> 2. In Peek make sure you have a keyboard shortcut set to start recording (default is `Ctrl` + `Alt` + `r`).
> 3. Using the configured keyboard shortcuts first Toggle Peek into fullscreen mode the start recording
> 4. To stop recording use Peek's keyboard shortcut again, get out of fullscreen again.
> 
> Of course this is something Peek could do on its own. My idea currently is that we add a "Fullscreen" checkbox to the new format selection popover. If checked Peek will switch into full screen mode before starting to record, and it will automatically unfullscreen when recording stops (so you get the UI back).
> 
> Of course this requires that Peek can set a global keyboard shortcut to stop recording. That means it is built with libkeybinder and the shortcut can be registered (doesn't conflict with any other global shortcut). I would just disable the fullscreen option if no keyboard shortcut is configured.

This didn't work for me, I can toggle fullscreen in other apps but not in peek :(

**Comment by matanster** [comment]
Same here. Peek won't toggle full-screen on my Ubuntu 20.04, although other apps do. It's so adorable though.

**Comment by jfernandz** [comment]
Also for Gnome 3.38.x there is not `Toggle fullscreen mode` shortcut. Besides I'm not using `gnome-shell` but just `gnome-flashback` because I'm using i3wm in combination with it (https://github.com/i3-gnome/i3-gnome).

**Comment by Xses-1** [comment]
KDE Plasma, still same issue as above

**Comment by orklann** [comment]
Love Peek, but record fullscreen (Full  Desktop) is what I need to record Blender, hope this issue will be fixed.

**Comment by jdmg94** [comment]
Love Peek but I'm facing the same issues with ElementaryOS' Pantheon and I can't record the top bar
